server/src/document/document.controller.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  Controller,
+  Post,
+  Req,
+  Res,
+  UseInterceptors,
+  UploadedFile,
+  HttpStatus,
+  HttpException,
+  Session,
+} from '@nestjs/common';
+import { ConfigService } from '../config/config.service';
+import { CrmService } from '../crm/crm.service';
+import { DocumentService } from './document.service';
+import { FileInterceptor } from '@nestjs/platform-express';
+import { Request } from 'express';
+
+@Controller('document')
+export class DocumentController {
+  constructor(
+    private readonly config: ConfigService,
+    private readonly crmService: CrmService,
+    private readonly documentService: DocumentService,
+) {}
+
+  /**
+   *  Upload a file to Sharepoint 
+   *
+   * @param      {FormData}  request  The request body should 
+   * of type FormData, have an 'instanceId' property and a
+   * 'file' property
+   * with the selected file blob.
+   * @return     {HTTP Response} Response
+   */
+  @Post('/')
+  @UseInterceptors(FileInterceptor('file'))
+  async index(@UploadedFile() file, @Req() request: Request, @Res() response, @Session() session) {
+    const {
+      body: {
+        instanceId
+      },
+    } = request;
+
+    const headers = {
+      // Document will be uploaded as this user
+      MSCRMCallerID: this.config.get('CRM_SERVICE_CONTACT_ID'),
+    };
+
+    let uploadDocResponse = {};
+
+    const encodedBase64File = Buffer.from(file.buffer).toString('base64');
+
+    const packageGUID = instanceId;
```

Comment:
Teeeeensy suggestion but you can do this on line 40: 

```
      body: {
        instanceId: packageGUID,
      },
```

though, idk, maybe it should just be instance id? reading this, I didn't assume it's specific to packages.